kite-tools-parent/kite-tools/src/test/java/org/kitesdk/cli/commands/TestCopyCommandCluster.java

```diff
@@ -142,16 +145,21 @@ public void testCopyWithoutCompaction() throws Exception {
     int size = DatasetTestUtilities.datasetSize(ds);
     Assert.assertEquals("Should contain copied records", 6, size);
 
-    Assert.assertEquals("Should produce 1 files",
-        1, Iterators.size(ds.pathIterator()));
+    Path[] paths = Iterators.toArray(ds.pathIterator(), Path.class);
+    Assert.assertEquals("Should produce " + expectedFiles + " files: " + Arrays.toString(paths),
+        expectedFiles, Iterators.size(ds.pathIterator()));
 
     verify(console).info("Added {} records to \"{}\"", 6l, dest);
     verifyNoMoreInteractions(console);
   }
 
   @Test
-  @SuppressWarnings("unchecked")
   public void testCopyWithNumWriters() throws Exception {
+    testCopyWithNumWriters(3);
+  }
+
+  @SuppressWarnings("unchecked")
+  public void testCopyWithNumWriters(int expectedFiles) throws Exception {
```

Comment:
I'm not thrilled with these test changes, but they drastically cut down on code duplication for the new tests that I added.